### PR TITLE
Feature- Notification Screen UI

### LINKED ISSUE #75: Notification Screen UI
**Complete By: 3rd May 2018, EOD**

This notification page will be opened from Navigation drawer where all the app notification will be shown.

**Features**:
- Notification types(as tabs): New Quiz, Deadlines, Resources
- For each tab:
    - use Recycler view to list the notification messages according to it’s type.
    - List item should have image,Notification title,description,date
    - Unread notifications should be on top of the list
    - Clear notification option - swipe to delete(single item) or long press to multi-select delete(multiple item).
- Notification Details page should have description and button to redirect to the required page based on the notification type.
    - Button Name : *Open Quiz* (new quiz and deadlines type) or *Open Resource* (resources type)

Sample UIs: https://github.com/UdacityAndroidDevScholarship/quiz-app/wiki/App-Mockups#notifications-activity
https://github.com/UdacityAndroidDevScholarship/quiz-app/wiki/App-Mockups#notification-detail-activity

### PR DESCRIPTION
**Pull request by nishank95**
Fixes #75 
Added Notifcation tabs UI and notification detail page.

## Description

This PR creates a Notification Screen UI containing three tabs (New Quiz,Deadline,Resources) and 
DetailScreen to show info about these types of Notifications.

## Screenshots

**BEFORE**:
N/A

**AFTER**:
<table style="border:'0px' ">
<tr>
<td>
new_quiz_tab<h3>TAB 1</h3>
</td>
<td>
deadlines_tab<h3>TAB 2</h3>
</td>
<td>
resources_tab<h3>TAB 3</h3>
</td>
</tr>
<tr>
<td>
new_quiz_details<h3>New Quiz Details</h3>
</td>
<td>
deadline_details<h3>Deadline Details</h3>
</td>
<td>
resources_details<h3>Resources Details</h3>
</td>
</tr>
</table>

## Other changes (e.g. bug fixes, UI tweaks, refactors)
Changed the icons representing different types of notifications

## Remaining Work
Swipe Delete Function( Swipe function (for item) added but as DB is not configured with app delete function is remaining)

### CODE REVIEW

**Comment by vansh1sh** [comment]
@nishank95 Please fix the latest conflicts with the branch.

**Comment by nishank95** [comment]
Resolved conflicts

File: app/build.gradle [hunk 1]
```diff
@@ -54,6 +54,7 @@ dependencies {
     implementation "com.google.firebase:firebase-database:$rootProject.firebaseVersion"
     implementation "com.google.firebase:firebase-messaging:$rootProject.firebaseVersion"
     implementation "com.android.support.constraint:constraint-layout:$rootProject.constraintLayoutVersion"
+    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
```
Comment (thayumaanavan) [inline app/build.gradle]: You can remove this line as constraint layout is already added in the above line.

**Comment by nishank95** [comment]
Thankyou @sagarsrao and @thayumaanavan

**Pull request merged**